### PR TITLE
Add Astrea quant quality tooling

### PR DESCRIPTION
**Pull request by Kaden-Schutt**
## Summary

This PR packages the quant-quality tooling from the Qwen3.5-9B MQ4 investigation so other hipfire contributors and agents can reproduce the loop instead of relying on ad hoc notes.

It adds:

- `scripts/astrea.py`, a JSON-first planning/reporting CLI for model inspection, calibration plans, quality metrics, dynamic tensor policy, promotion recipes, KV-policy sketches, bundle plans, and evidence reports.
- `.agents/skills/astrea/`, an agent skill that captures the workflow contract and guardrails for quant calibration, MoE ingress, KV policy, and Atlas handoff.
- Qwen3.5 PyTorch oracle/debug scripts for hidden/logits comparison, HFQ tensor attribution, projection-error ranking, and narrow engine-vs-HFQ replay probes.
- `dump_qwen35_hidden` and `probe_qwen35_l0_ops` runtime examples to export hipfire tensors for PyTorch oracle comparison.
- Non-finite logit checks in `eval_hipfire` so invalid candidates fail instead of entering the KLD ledger.
- Compact experiment evidence under `experiments/quant_fix_20260513/`: ledger, current policy map, current summary, and the PyTorch oracle POC note. Large raw dumps/model artifacts are intentionally not committed.
- Methodology docs in `docs/methodology/`, especially `quant-quality-tooling.md`, explaining the runbook, commands, evidence contract, what worked, and what did not.

## How we used it

The first Qwen3.5-9B MQ4 run used ROCm PyTorch on `hiptrx` as the oracle and hipfire's BF16 KLD reference as the fixed scoring target.

Fixed 20-chunk eval contract:

```bash
HIPFIRE_NORMALIZE_PROMPT=0 HIPFIRE_GRAPH=0 HIPFIRE_KV_MODE=asym3 \
./target/release/examples/eval_hipfire \
  --model CANDIDATE.hfq \
  --ref benchmarks/quality-baselines/refs/qwen3.5-9b-bf16.kldref.bin \
  --output experiments/quant_fix_20260513/eval/CANDIDATE.prefill-c20.kldseq \
  --scoring-mode prefill \
  --max-chunks 20
```

Measured progression:

| Artifact | 20-chunk KLD | PPL | Eval tok/s | Status |
|---|---:|---:|---:|---|
| baseline MQ4 | `0.330882` | `9.3404` | `346` | baseline |
| prior uniform LS/clip POC | `0.308592` | `9.3769` | n/a | rejected: PPL regression |
| `cand143` | `0.210238` | `9.2806` | `328` | previous best, full hidden/logits oracle completed |
| `cand148` | `0.206809` | `9.2878` | `328` | current KLD/PPL/speed best, full oracle/decode validation still pending |
| full-MQ6 + q8 KV control | `0.067687` | `9.3978` | `233` | proves sub-0.09 KLD is visible, but fails PPL/speed |

Key finding: the current simple tensor-selective MQ4/MQ6 loop is flattening around `0.205-0.21`, but that is not a theoretical floor. The full-MQ6/q8 control proves the harness can see Q4-class KLD. The next likely lever is GPTQ/Hessian-aware block refit or a stronger rotated-domain objective, not more local rel-RMSE ranking.

## How this helps others

A contributor can now:

1. fingerprint the engine and model with Astrea;
2. dump hipfire hidden/logits tensors for a candidate;
3. compare the dump against HF/PyTorch BF16;
4. rank tensors by attribution/projection error;
5. write a candidate policy or promotion recipe;
6. evaluate KLD/PPL under a fixed contract;
7. record accept/reject decisions in compact artifacts;
8. hand runtime-sensitive candidates to Atlas for AR/DFlash perf validation.

The docs explicitly separate proof from hypotheses: local attribution ranks are candidate generators, not acceptance criteria.

## Validation

Ran on `hiptrx` from the dirty development worktree before commit:

```bash
/home/kaden/.venvs/hipfire-rocm-torch-amd721/bin/python3 scripts/test_astrea.py
cargo build --release --features deltanet \
  --example eval_hipfire \
  --example probe_qwen35_l0_ops \
  --example dump_qwen35_hidden \
  -p hipfire-runtime
git diff --cached --check
```

Results:

- `scripts/test_astrea.py`: 34/34 passing.
- release build: passed, warnings only.
- whitespace check: passed.

Commit hook also ran:

- short coherence battery: no hard errors, report `/tmp/coherence-20260513-214107.md`.
- agentic gate: pass with 1 soft warning, report `/tmp/agentic-gate-20260513-214127.md`.
- MQ4 speed gate: no metrics checked because the 4B model was absent.

Then verified again from a clean detached worktree at commit `4e6ca73d`:

```bash
/home/kaden/.venvs/hipfire-rocm-torch-amd721/bin/python3 scripts/test_astrea.py
cargo build --release --features deltanet \
  --example eval_hipfire \
  --example probe_qwen35_l0_ops \
  --example dump_qwen35_hidden \
  -p hipfire-runtime
```

Results:

- `scripts/test_astrea.py`: 34/34 passing.
- clean-worktree release build: passed, warnings only.

## Notes

The hiptrx worktree still contains unrelated uncommitted MoE/Q8 prefill changes in `crates/hipfire-arch-qwen35/src/qwen35.rs` and large raw local experiment artifacts. They are intentionally excluded from this PR.

### CODE REVIEW

**Comment by Kaden-Schutt** [comment]
Closing as superseded — the live Astrea tooling (`scripts/astrea.py`, agent skill, `docs/methodology/astrea-model-policy.md`) landed on master via direct push (`f222c0ac`). The remaining files in this PR are a frozen per-experiment snapshot of the `quant_fix_20260513` paroquant-v4 run plus diagnostic probe binaries; not worth maintaining as an open PR. Probe binaries can be re-introduced as standalone PRs if needed for future quant work.
